### PR TITLE
Show new invitation quota on `ProposalDetails` (#3902)

### LINKED ISSUE #3902: [USERSNAP] Proposal does not display the necessary information
**Taken at**: https://dao.joystream.org/#/proposals/preview/40
**Fields**:
- Houston, we have a problem.: Proposal does not display the necessary information
- Comment: Set Membership Lead Invitation Quota Proposal.
Proposal for 100 invites.  The proposal, for some unknown reason, does not display this information.
- Feedback: adovrn

SCREENSHOT
*[Download original image](https://resources.usersnap.com/company/f9f719df-a929-4461-bb6e-46dbfbceffb7/datapoint_screenshot/a11b689a-885e-4ab1-8f94-11d7bbe9aa69-annotated_dd308136-a45c-40e6-93b3-1747a9b53a56.png?etag=74a86b4b02667dad57137cfe9b2e295c)*

**Browser**: Chrome 107 (Windows 10)
**Screen size**: 1536x864
**Browser size**: 1536x792
**[Open #346 in Usersnap](https://app.usersnap.com/l/feedback/894a3d1e-150e-45d1-836f-fe8f9a4fd9ec?utm_medium=integration&utm_campaign=integration&utm_content=open)**

Powered by **[Usersnap](https://usersnap.com/?utm_source=product&amp;utm_medium=poweredbylink&amp;utm_campaign=github_entry)**.

### PR DESCRIPTION
**Pull request by traumschule**
Fixes #3902

### CODE REVIEW

**Comment by oleksanderkorn** [comment]
@traumschule what do you mean by `parameter doesn't show up.` ? 
I see it on the preview branch:
image

**Comment by oleksanderkorn** [comment]
@traumschule the proposalDetails are not defined for such type:
image

The trick is the missing definition of `SetMembershipLeadInvitationQuotaDetails`:
image
See for example `const asSetReferralCut: DetailsCast<'SetReferralCutProposalDetails'> = (fragment): SetReferralCutDetails => ({`
It is autogenerated from graphql schema, same should be done here for our type `setMembershipLeadInvitationQuota`
image

**Comment by oleksanderkorn** [comment]
image

**Comment by oleksanderkorn** [comment]
@traumschule check this out https://github.com/traumschule/pioneer/pull/4/files :)

**Comment by traumschule** [comment]
Thanks!

count

**Review by oleksanderkorn — COMMENTED** [review]
If we change type to count should it also be the count on proposal creation? Because currently this is treat as `amount`:

image

**Review by oleksanderkorn — APPROVED** [review]

**Review by thesan — APPROVED** [review]
LGTM but I don't really understand how it fixes #3902 :confused:

File: packages/ui/src/proposals/types/ProposalDetails.ts [hunk 1]
```diff
@@ -298,6 +298,13 @@ const asSetReferralCut: DetailsCast<'SetReferralCutProposalDetails'> = (fragment
   amount: new BN(fragment.newReferralCut),
 })
 
+const asSetMembershipLeadInvitationQuota: DetailsCast<'SetMembershipLeadInvitationQuotaProposalDetails'> = (
+  fragment
+): SetMembershipLeadInvitationQuota => ({
+  type: 'setMembershipLeadInvitationQuota',
+  amount: new BN(fragment.newLeadInvitationQuota),
+})
+
```
Comment (traumschule) [inline packages/ui/src/proposals/types/ProposalDetails.ts:307]: the key was to cast the detail

**Pull request merged**